### ISSUE TITLE
The `Use SQL to limit fetch size` is not working for Databend

### ISSUE THREAD

**Issue by pengye91** [dbeaver/dbeaver#23129]
### Description

Hi team,

I'm using DBeaver Community Edition 24.0.0 and trying to use a custom [Databend JDBC driver](https://repo1.maven.org/maven2/com/databend/databend-jdbc/0.1.9/).

All works fine except that the  `Use SQL to limit fetch size` checkbox seems not working.

as the screenshots shows, the LIMIT clause is not attached to the query automatically.

<img width="984" alt="image" src="https://github.com/dbeaver/dbeaver/assets/17252843/3923c2fd-afca-450e-931e-7c317499d594">
<img width="619" alt="image" src="https://github.com/dbeaver/dbeaver/assets/17252843/34bed254-d3bf-4a2d-8a00-b35f6a06ebd1">

And I tested with a internal MySQL driver and it turned out the LIMIT clause is attached to the query automatically, so I assume that this is something wrong with the custom driver.

can someone kindly show me how this LIMIT clause is attached automatically and how I can improve the custom driver to implement that?

thank you.

### DBeaver Version

Community Edition 24.0.0

### Operating System

macOS Sonoma Version 14.1.1 Apple M1 Max Chip

### Database and driver

Database: Databend https://www.databend.com/

driver: https://repo1.maven.org/maven2/com/databend/databend-jdbc/0.1.9/

**Comment by LonwoLonwo** [comment]
Hello @pengye91 

This setting doesn't have a default query syntax for the query limitation. Because different databases have different standards. So, we need to implement it first for your database. Which we can't do without the Databend extension.

Like this:

```java
public DBCQueryTransformer createQueryTransformer(@NotNull DBCQueryTransformType type) {
        if (type == DBCQueryTransformType.RESULT_SET_LIMIT) {
            return new QueryTransformerLimitVertica();
        }
        return null;
}
```

**Comment by pengye91** [comment]
Hi @LonwoLonwo,

Got it, so that means currently we can't do anything until the Databend extension is merged into the codebase (like #19558  ), right?

**Comment by LonwoLonwo** [comment]
Yes, you are correct. But. By default, if you have any limitations settings (resultset = 200, as an example), DBeaver sends this to the driver:
`dbStat.setLimit(offset, maxRows);` which executes -> `Statement#setMaxRows`
So, if your driver has this implementation, it should work also.

**Comment by pengye91** [comment]
Great, that's what I need, thank you so much.
